posts/2024-08-26-webassembly-targets-and-new-on-by-default-features.md

```diff
@@ -0,0 +1,191 @@
+---
+layout: post
+title: "WebAssembly targets and new on-by-default features"
+author: Alex Crichton
+---
+
+The Rust compiler has [recently upgraded to using LLVM 19][llvm19] and this
+change accompanies some updates to WebAssembly targets of the Rust compiler.
+Nightly Rust, what will be come Rust 1.82 on 2024-10-17, reflects all of these
+changes and can be used for testing.
+
+WebAssembly is an evolving standard where new features are being added over time
+through a [proposals process][proposals]. As WebAssembly proposals reach
+maturity, get merged into the specification itself, get implemented in engines,
+and remains this way for quite some time then producer toolchains (e.g. LLVM)
+are going to update to include these new proposals by default. In LLVM 19 this
+has happened with the [multi-value and reference-types proposals][llvmenable].
+These are now enabled by default in LLVM and transitively means that it's
+enabled by default for Rust as well.
+
+WebAssembly targets for Rust now [have improved
+documentation](https://github.com/rust-lang/rust/pull/128511) about WebAssembly
+features and disabling them, and this post is going to review these changes and
+go into depth about what's changing in LLVM.
+
+## Enabling Reference Types by Default
+
+The [reference-types proposal to
+WebAssembly](https://github.com/webAssembly/reference-types) introduced a few
+new concepts to WebAssembly, notably the `externref` type which is a
+host-defined GC resource that WebAssembly cannot access but can pass around.
+Rust does not have support for the WebAssembly `externref` type and LLVM 19 does
+not change that. WebAssembly modules produced from Rust will continue to not use
+the `externref` type nor have a means of being able to do so.
```

Comment:
Question: is this going to *always* be the case, or for the foreseeable future?

Reply:
I'll clarify that there are no plans for this but even if it were done it likely wouldn't affect modules by default.